rex/crash.py

```diff
@@ -633,13 +633,14 @@ def _triage_crash(self):
 
         # grab the all actions in the last basic block
         symbolic_actions = [ ]
-        for a in self.prev.history.recent_actions + self.state.history.recent_actions:
+        for a in list(self.prev.history.actions) + list(self.state.history.actions):
```

Comment:
I don't think this is a good change - `actions` is in fact an iterator over the contents of `recent_actions` on itself and each of its parents, so this will duplicate a bunch of the history.

Reply:
This is, of course, assuming that self.prev and self.state share any history.

I'm not sure what the original intent was (this is the main problem with trying to debug rex), but if this change fixed bugs and they do share history then the correct version is just `for a in self.state.history.actions`, assuming that self.state is a descendant of self.prev.

Reply:
Well, the reason why it is `actions` instead of `recent_actions` is that `recent_actions` does not contain actions from the parent (anymore, I guess?), so the list will always be empty. I've made it `list(actions)` just to be sure that it works because I am a noob in Python and don't know if `iterator` can be `+`... Sorry about it :(

Reply:
And I don't know about whether `prev` and `state` share any history. Not sure why we have to add the 2 lists though, will have to wait for @salls to shed some light on this

Reply:
Apparently what salls really wants is all actions but only in 2 last blocks. The problem with `recent_actions` is that it is not copied from parent, and at each step, `new_state.history = old_state.history.make_child()`. So basically, `recent_actions` is always empty. That's why I used `actions`. But it is not exactly what salls wants either. It does fix some test cases, but not all of them. `actions` is not empty anymore but it contains too many elements, and not in the right order... What would be the way to get the list of 2 actions: the first one is the action from the block before the last block then the second one is the action from the last block? Because their order in `symbolic_actions` matters: it will determine the crash type... You implemented the new `history` iterators @rhelmot, can you shed some light on this please?